osu.Game.Rulesets.Osu/Mods/OsuModRelax.cs

```diff
@@ -38,19 +38,25 @@ public class OsuModRelax : ModRelax, IUpdatableByPlayfield, IApplicableToDrawabl
         private ReplayState<OsuAction> state = null!;
         private double lastStateChangeTime;
 
+        private DrawableOsuRuleset ruleset = null!;
+
         private bool hasReplay;
+        private bool legacyReplay;
 
         public void ApplyToDrawableRuleset(DrawableRuleset<OsuHitObject> drawableRuleset)
         {
+            ruleset = (DrawableOsuRuleset)drawableRuleset;
+
             // grab the input manager for future use.
-            osuInputManager = ((DrawableOsuRuleset)drawableRuleset).KeyBindingInputManager;
+            osuInputManager = ruleset.KeyBindingInputManager;
         }
 
         public void ApplyToPlayer(Player player)
         {
             if (osuInputManager.ReplayInputHandler != null)
             {
                 hasReplay = true;
+                legacyReplay = ruleset.ReplayScore.ScoreInfo.IsLegacyScore;
```

Comment:
Hmmmmm.....

On one side, I could get behind this in a sense, as this is a rather simple way to address this. That said, the other way to go with this would be to force lazer to output replay frames even for relax scores. Doing that would mean that if relax input generation is somehow changed in the future, old replays aren't broken, which is probably better in the long run.

@ppy/team-client any strong opinions? not sure I have one. Maybe this is okay as a stopgap at least?

Reply:
> the other way to go with this would be to force lazer to output replay frames even for relax scores.

I'm not entirely sure what you mean by this. For lazer replays, this is already the case. It's only legacy replays which do not have key-presses in the replay data.

Reply:
Yeah I got confused and edited my comment sorry.

Reply:
This looks pretty okay to me.